Lesson19/Ece/script.js

```diff
@@ -19,35 +18,62 @@
 const newParagraph = document.createElement("p");
-newParagraph.innerHTML="Lorem ipsum dolor <a href=\"https://dsds.dsds\">sit amet consectetur </a>adipisicing elit. Illum esse, recusandae sint error possimus hic, voluptates vel eveniet est, deleniti reprehenderit fugit minus? Officiis iusto, incidunt molestiae neque quas omnis.";
+newParagraph.innerHTML =
+  'Lorem ipsum dolor <a href="https://dsds.dsds">sit amet consectetur </a>adipisicing elit. Illum esse, recusandae sint error possimus hic, voluptates vel eveniet est, deleniti reprehenderit fugit minus? Officiis iusto, incidunt molestiae neque quas omnis.';
 document.body.appendChild(newParagraph);
 
-const firstParagraph = document.getElementById('first-paragraph');
+const firstParagraph = document.getElementById("first-paragraph");
 // If it was a list, you could use the code below to insert the new element
 // firstParagraph.insertBefore(newParagraph, positionWhereToInsert);
 
 // 4. Remove the second image from the page.
-const images = document.body.querySelectorAll('img');
+const images = document.body.querySelectorAll("img");
 console.log(images);
 images[1].remove();
 
 // 5. Add a border to all images on the page (2px solid #272932).
+const allImages = document.body.querySelectorAll("img");
+console.log("allImages");
+allImages.style.border = "2px solid #272932";
+
 // 6. Change the text of the first heading on the page to "Welcome to my Website".
+const firstHeading = document.getElementsByTagName("h1");
+firstHeading.textContent = "Welcome to my Website";
+
 // 7. Replace the first image on the page with a new image ("https://picsum.photos/400/200").
-// 8. Remove the first list item from the unordered list.
-// 9. Append a new list item to the end of the ordered list with the text "Last Item".
-// 10. Add a button (in html) with "Hide images" text. When a button is clicked, change the text of the button to "Show images" and hide all the images on the web page (set display to none).
+const firstImage = document.getElementById("img1");
+firstImage.src = "https://picsum.photos/400/200";
 
+// 8. Remove the first list item from the unordered list.
+const listItems = document.body.querySelectorAll("li");
+console.log(listItems);
+listItems[1].remove();
 
+// 9. Append a new list item to the end of the ordered list with the text "Last Item".
+const newListItem = document.createElement("li");
+const newListItemText = document.createTextNode("Last Item");
+newListItem.appendChild(newListItemText);
+document.getElementById("ul").appendChild(newListItem);
 
+// 10. Add a button (in html) with "Hide images" text. When a button is clicked, 
+//change the text of the button to "Show images" and 
+//hide all the images on the web page(set display to none).
+const newButton = document.createElement("button");
```

Comment:
When you create an element in JS, you need to append to an element that exists in the HTML. Right now, the newButton is not visible on the page as it's not appended anywhere.
Since you use `id` inside the `showImages` function, you also need to add it to the newly created element.

I suggest code changes like this for the code from line 72 until the end of the file.
```
const newButton = document.createElement("button");
newButton.id = 'button';
newButton.textContent = "Hide images";
newButton.addEventListener("click", hideImages);
document.getElementById('main').appendChild(newButton);

function hideImages(){
  document.getElementById('button').innerText = "Show images";
  // all images were selected before so now you can use them directly
  allImages.forEach(image => image.style.display = "none");
};
```